Tasmanian+rocm testing build fails on frontier

I'm testing Tasmanian built with spack on Frontier for E4S validation using the following spec:
Edit: (Targeting frontier, these results are actually from crusher)
```
-- linux-sles15-zen3 / gcc@11.2.0 -------------------------------
nopqpbc tasmanian@7.9~blas~cuda~fortran~ipo~magma+mpi~openmp~python+rocm~xsdkflags amdgpu_target=gfx90a build_system=cmake build_type=Release
7o5iua6     cmake@3.25.2~doc+ncurses+ownlibs~qt build_system=generic build_type=Release
igbrz2c         ncurses@6.3~symlinks+termlib abi=none build_system=autotools
savxweu             pkgconf@1.8.0 build_system=autotools
gv7ah47         openssl@1.1.1-os~docs~shared build_system=generic certs=mozilla
vw5wbnb     cray-mpich@8.1.23+wrappers build_system=generic
5fmb57z     hip@5.4.0~cuda~ipo+rocm build_system=cmake build_type=Release patches=ca523f1
euzfcae     hsa-rocr-dev@5.4.0+image~ipo+shared build_system=cmake build_type=Release patches=71e6851
z5r2sar     llvm-amdgpu@5.4.0~ipo~link_llvm_dylib~llvm_dylib~openmp+rocm-device-libs build_system=cmake build_type=Release patches=a08bbe1
x4yi2g3     rocblas@5.4.0~ipo+tensile amdgpu_target=auto build_system=cmake build_type=Release patches=81591d9
cpdjwvl     rocsolver@5.4.0~ipo+optimal amdgpu_target=auto build_system=cmake build_type=Release patches=8067bfb
icwqurs     rocsparse@5.4.0~ipo~test amdgpu_target=auto build_system=cmake build_type=Release
```

I copied the testing directory from <tasmanian root>/share/tasmanian/testing, did `mkdir build;cd build; cmake ..` and that worked but make failed with the error output below. Any advice on how to get this working? I'm happy to dig up any additional diagnostic info that could help.

```
[ 54%] Linking CXX executable example_sparse_grids
Tasmanian Sparse Grids  version: 7.9
                        license: BSD 3-Clause with UT-Battelle disclaimer
          OpenMP multithreading: Disabled
          GPU backend framework: ROCm/HIP
         Available acceleration:  gpu-cublas gpu-cuda
                 Available GPUs:
          0:                     with  65520MB of RAM
          1:                     with  65520MB of RAM

/usr/lib64/gcc/x86_64-suse-linux/7/../../../../x86_64-suse-linux/bin/ld: CMakeFiles/example_sparse_grids.dir/example_sparse_grids_05.cpp.o: undefined reference to symbol 'pthread_create@@GLIBC_2.2.5'
/usr/lib64/gcc/x86_64-suse-linux/7/../../../../x86_64-suse-linux/bin/ld: /lib64/libpthread.so.0: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
make[2]: *** [examples_cxx/CMakeFiles/example_sparse_grids.dir/build.make:282: examples_cxx/example_sparse_grids] Error 1
make[1]: *** [CMakeFiles/Makefile2:102: examples_cxx/CMakeFiles/example_sparse_grids.dir/all] Error 2
make: *** [Makefile:101: all] Error 2

```

@mkstoyanov I'm seeing the same error with tasmanian@develop, both +openmp and ~openmp:

```
[wspear@login2.crusher build]$ make
[  4%] Building CXX object examples_cxx/CMakeFiles/example_sparse_grids.dir/example_sparse_grids_01.cpp.o
[  8%] Building CXX object examples_cxx/CMakeFiles/example_sparse_grids.dir/example_sparse_grids_02.cpp.o
[ 12%] Building CXX object examples_cxx/CMakeFiles/example_sparse_grids.dir/example_sparse_grids_03.cpp.o
[ 16%] Building CXX object examples_cxx/CMakeFiles/example_sparse_grids.dir/example_sparse_grids_04.cpp.o
[ 20%] Building CXX object examples_cxx/CMakeFiles/example_sparse_grids.dir/example_sparse_grids_05.cpp.o
[ 25%] Building CXX object examples_cxx/CMakeFiles/example_sparse_grids.dir/example_sparse_grids_06.cpp.o
[ 29%] Building CXX object examples_cxx/CMakeFiles/example_sparse_grids.dir/example_sparse_grids_07.cpp.o
[ 33%] Building CXX object examples_cxx/CMakeFiles/example_sparse_grids.dir/example_sparse_grids_08.cpp.o
[ 37%] Building CXX object examples_cxx/CMakeFiles/example_sparse_grids.dir/example_sparse_grids_09.cpp.o
[ 41%] Building CXX object examples_cxx/CMakeFiles/example_sparse_grids.dir/example_sparse_grids_10.cpp.o
[ 45%] Building CXX object examples_cxx/CMakeFiles/example_sparse_grids.dir/example_sparse_grids_11.cpp.o
[ 50%] Building CXX object examples_cxx/CMakeFiles/example_sparse_grids.dir/example_sparse_grids.cpp.o
[ 54%] Linking CXX executable example_sparse_grids
Tasmanian Sparse Grids  version: 7.10 (development)
                git commit hash: e62d47ed20b44160ce963263a6fc9000deadb83c
                cmake cxx flags: Release, 
                        license: BSD 3-Clause with UT-Battelle disclaimer
          OpenMP multithreading: Disabled
          GPU backend framework: ROCm/HIP
         Available acceleration:  gpu-cublas gpu-cuda
                 Available GPUs:
          0:                     with  65520MB of RAM
          1:                     with  65520MB of RAM

/usr/bin/ld: CMakeFiles/example_sparse_grids.dir/example_sparse_grids_05.cpp.o: undefined reference to symbol 'pthread_create@@GLIBC_2.2.5'
/usr/bin/ld: /lib64/libpthread.so.0: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
make[2]: *** [examples_cxx/CMakeFiles/example_sparse_grids.dir/build.make:282: examples_cxx/example_sparse_grids] Error 1
make[1]: *** [CMakeFiles/Makefile2:102: examples_cxx/CMakeFiles/example_sparse_grids.dir/all] Error 2
make: *** [Makefile:101: all] Error 2

```

--- a/var/spack/repos/builtin/packages/tasmanian/package.py
+++ b/var/spack/repos/builtin/packages/tasmanian/package.py
@@ -189,12 +189,13 @@ class Tasmanian(CMakePackage, CudaPackage, ROCmPackage):
 
         options = [cmake_dir]
         if "+rocm" in self.spec:
+            options.append(f"-Dhip_DIR={self.spec['hip'].prefix.lib.cmake.hip}")
             options.append(
                 f"-DAMDDeviceLibs_DIR={self.spec['llvm-amdgpu'].prefix.lib.cmake.AMDDeviceLibs}"
             )
             options.append(f"-Damd_comgr_DIR={self.spec['comgr'].prefix.lib.cmake.amd_comgr}")
             options.append(
-                f"-Dhsa-runtime64_DIR={self.spec['hsa-rocr-dev'].prefix.lib.cmake.hsa-runtime64}"
+                f"-Dhsa-runtime64_DIR={join_path(self.spec['hsa-rocr-dev'].prefix.lib.cmake, 'hsa-runtime64')}"
             )
             options.append(f"-DHSA_HEADER={self.spec['hsa-rocr-dev'].prefix.include}")
             options.append(f"-DCMAKE_INCLUDE_PATH={self.spec['hsa-rocr-dev'].prefix.include.hsa}")

--- a/var/spack/repos/builtin/packages/tasmanian/package.py
+++ b/var/spack/repos/builtin/packages/tasmanian/package.py
@@ -195,7 +195,8 @@ class Tasmanian(CMakePackage, CudaPackage, ROCmPackage):
             )
             options.append(f"-Damd_comgr_DIR={self.spec['comgr'].prefix.lib.cmake.amd_comgr}")
             options.append(
-                f"-Dhsa-runtime64_DIR={join_path(self.spec['hsa-rocr-dev'].prefix.lib.cmake, 'hsa-runtime64')}"
+                "-Dhsa-runtime64_DIR=" +
+                join_path(self.spec['hsa-rocr-dev'].prefix.lib.cmake, 'hsa-runtime64')
             )
             options.append(f"-DHSA_HEADER={self.spec['hsa-rocr-dev'].prefix.include}")
             options.append(f"-DCMAKE_INCLUDE_PATH={self.spec['hsa-rocr-dev'].prefix.include.hsa}")

--- a/var/spack/repos/builtin/packages/tasmanian/package.py
+++ b/var/spack/repos/builtin/packages/tasmanian/package.py
@@ -195,8 +195,8 @@ class Tasmanian(CMakePackage, CudaPackage, ROCmPackage):
             )
             options.append(f"-Damd_comgr_DIR={self.spec['comgr'].prefix.lib.cmake.amd_comgr}")
             options.append(
-                "-Dhsa-runtime64_DIR=" +
-                join_path(self.spec['hsa-rocr-dev'].prefix.lib.cmake, 'hsa-runtime64')
+                "-Dhsa-runtime64_DIR="
+                + join_path(self.spec['hsa-rocr-dev'].prefix.lib.cmake, 'hsa-runtime64')
             )
             options.append(f"-DHSA_HEADER={self.spec['hsa-rocr-dev'].prefix.include}")
             options.append(f"-DCMAKE_INCLUDE_PATH={self.spec['hsa-rocr-dev'].prefix.include.hsa}")

--- a/var/spack/repos/builtin/packages/tasmanian/package.py
+++ b/var/spack/repos/builtin/packages/tasmanian/package.py
@@ -196,7 +196,7 @@ class Tasmanian(CMakePackage, CudaPackage, ROCmPackage):
             options.append(f"-Damd_comgr_DIR={self.spec['comgr'].prefix.lib.cmake.amd_comgr}")
             options.append(
                 "-Dhsa-runtime64_DIR="
-                + join_path(self.spec['hsa-rocr-dev'].prefix.lib.cmake, 'hsa-runtime64')
+                + join_path(self.spec["hsa-rocr-dev"].prefix.lib.cmake, "hsa-runtime64")
             )
             options.append(f"-DHSA_HEADER={self.spec['hsa-rocr-dev'].prefix.include}")
             options.append(f"-DCMAKE_INCLUDE_PATH={self.spec['hsa-rocr-dev'].prefix.include.hsa}")

